### PR TITLE
fix: clippy warnings in tests

### LINKED ISSUE #1462: fix all clippy warnings
### Describe the feature

there are clippy warnings:

```
cargo +nightly clippy --all --all-features
```

## TODO

fix all obvious clippy warnings

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by thor314**
Had a minute and this looked easy, closes #1462.

### CODE REVIEW

**Review by onbjerg — APPROVED** [review]
Thank u sir, lgtm after CI

File: crates/consensus/beacon/src/engine/mod.rs [hunk 1]
```diff
@@ -767,7 +775,7 @@ mod tests {
             let block1 = random_block(1, Some(genesis.hash), None, Some(0));
             insert_blocks(env.db.as_ref(), [&genesis, &block1].into_iter());
 
-            let _ = spawn_consensus_engine(consensus_engine);
+            let _ = spawn_consensus_engine(consensus_engine).await;
```
Comment (rkrasiuk) [inline crates/consensus/beacon/src/engine/mod.rs:778]: the consensus engine is a future that shouldn't exit when ignored. to fix the warning you prob can keep the receiver returned from spawn and drop it at the end of the tests

**Review by rkrasiuk — CHANGES_REQUESTED** [review]

File: crates/consensus/beacon/src/engine/mod.rs (hunk 1 above)
Comment (mattsse) [reply crates/consensus/beacon/src/engine/mod.rs:778]: this awaits on the oneshot response though
which does not seem to receive anything

imo it is reasonable to await the receiver return value.

**Review by rkrasiuk — APPROVED** [review]
lgtm, ty!

**Review by mattsse — APPROVED** [review]

**Pull request merged**